techniques/failures/F97.html

```diff
@@ -0,0 +1,123 @@
+   match for a "coarse" primary input mechanism */
+
+if (window.matchMedia && window.matchMedia("(pointer:coarse)").matches) {
+  /* set up event listeners for touch */
+  target.addEventListener('touchend', ...);
+  ...
+} else {
+  /* set up event listeners for mouse/keyboard */
+  target.addEventListener('click', ...);
+  ...
+}
+				</code></pre>
+			
+				<p>Similarly, web content that omits relevant/necessary keyboard event listeners (e.g. for the correct keyboard interaction with a complex widget, such as a tab interface) when a touchscreen is detected - under the assumption that on a touch device, keyboard support won't be necessary.</p>
+
+				<pre><code>
+/* inferring the presence of a touchscreen based on
+   the navigator.maxTouchPoints property defined in
+   the Pointer Events API */
+
+if (!(window.PointerEvent && ('maxTouchPoints' in navigator) &&
+    (navigator.maxTouchPoints > 0))) {
+  /* set up event listeners for keyboard interactions */
+  target.addEventListener('keyup', ...);
+  ...
+} else {
+  /* no need to listen to keyboard - there's a touchscreen... */
+  ...
+}
+				</code></pre>
+			
+				<p class="note">Generally, these approaches will <em>also</em> result in a failure of Success Criterion 2.1.1 Keyboard - but only in situations where a touchscreen interface was also detected.</p>
+
+			</section>
+				
+      <section class="example">
+				<h3>Hiding/omitting controls for mouse users when a touchscreen is detected</h3>
+				<p>Web content containing interactive widgets such as content carousels, with visible buttons to operate the widget (such as previous/next buttons, or a visible scrollbar/slider). These visible controls are hidden/omitted when a touchscreen is detected, under the assumption that users will simply use touch gestures to operate the widgets.</p>
+        <pre><code>
+/* using CSS Media Queries 4 Interaction Media Features
+   to hide particular elements in the page (such as a container
+   with visible controls) when a "coarse" primary input is present */
+
+@media (pointer: coarse) {
+  #widget .controls { display: none; }
+}
+				</code></pre>
+				
+				<p class="note">Generally, these approaches will <em>also</em> result in a failure of Success Criterion 2.1.1 Keyboard <em>and</em> (depending on the touch gesture that the user is expected to perform) Success Criterion 2.5.1 Pointer Gestures.</p>
+
+      </section>
+
+    </section>
+    
+		<section id="tests">
+			<h2>Tests</h2>
+			<section class="test-procedure">
+				<h3>Procedure</h3>
+				<ol>
+					<li>Open the content on a device with a touchscreen and at least one additional input modality - this includes touch-enabled laptops and touchscreen devices (smartphones or tablets) with a paired external keyboard and mouse.
+					<li>Check that all interactive controls (such as links, form inputs, or complex custom widgets) are still visible/present (compared to the same content when viewed on a device without a touchscreen)</li>
```

Comment:
Is this actually required for this failure?

Reply:
good catch. i reworded the procedure step (and expanded the last examples explanatory text slightly), as it's not a failure to hide things per se, but not then providing something that CAN be operated with an input other than touch.